lib/msprime.c

```diff
@@ -5823,6 +5823,105 @@ msp_add_mass_migration(
     return ret;
 }
 
+/* Population split */
+
+static int
+msp_population_split(msp_t *self, demographic_event_t *event)
+{
+    int ret = 0;
+    population_id_t *derived = event->params.population_split.derived;
+    population_id_t ancestral = event->params.population_split.ancestral;
+    size_t num_populations = event->params.population_split.num_derived;
+    demographic_event_t mass_migration;
+    size_t j, k;
+    size_t N = self->num_populations;
+
+    /* The mass migration moves lineages into the ancestral population */
+    mass_migration.params.mass_migration.destination = ancestral;
+    mass_migration.params.mass_migration.proportion = 1.0;
+
+    for (j = 0; j < num_populations; j++) {
+        /* Turn off all migration to and from derived[j] */
+        for (k = 0; k < self->num_populations; k++) {
+            self->migration_matrix[((size_t) derived[j] * N) + k] = 0;
+            self->migration_matrix[k * N + (size_t) derived[j]] = 0;
+        }
```

Comment:
Turning off the migrations here is nice. How about also setting the population sizes to zero for the derived populations? That should make it harder to accidentally reuse those population IDs when the populations ought not exist.

Reply:
Good idea. Let's see how this interacts with start_time/end_time, might not be necessary.